cmd/kubeadm/app/cmd/phases/uploadconfig.go

```diff
@@ -40,49 +40,109 @@ var (
 		This enables correct configuration of system components and a seamless user experience when upgrading.
 
 		Alternatively, you can use kubeadm config.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.KubeadmConfigConfigMap, metav1.NamespaceSystem)
+		`), kubeadmconstants.KubeadmConfigConfigMap, metav1.NamespaceSystem)
 
 	uploadConfigExample = normalizer.Examples(`
 		# uploads the configuration of your cluster
 		kubeadm alpha phase upload-config --config=myConfig.yaml
 		`)
+
+	kubeletConfigUploadLongDesc = normalizer.LongDesc(`
+		Uploads kubelet configuration extracted from the kubeadm InitConfiguration object to a ConfigMap
+		of the form kubelet-config-1.X in the cluster, where X is the minor version of the current (API Server) Kubernetes version.
+		`)
+
+	kubeletConfigUploadExample = normalizer.Examples(`
+		# Uploads the kubelet configuration from the kubeadm Config file to a ConfigMap in the cluster.
+		kubeadm init phase config-upload kubelet --config kubeadm.yaml
+		`)
 )
 
-// NewCmdUploadConfig returns the Cobra command for running the uploadconfig phase
-func NewCmdUploadConfig() *cobra.Command {
-	cfg := &kubeadmapiv1beta1.InitConfiguration{}
-	kubeConfigFile := kubeadmconstants.GetAdminKubeConfigPath()
-	var cfgPath string
-
-	cmd := &cobra.Command{
-		Use:     "upload-config",
-		Short:   "Uploads the currently used configuration for kubeadm to a ConfigMap",
-		Long:    uploadConfigLongDesc,
-		Example: uploadConfigExample,
+// uploadConfigData defines the behavior that a runtime data struct passed to the uploadConfig phase should
+// have. Please note that we are using an interface in order to make this phase reusable in different workflows
+// (and thus with different runtime data struct, all of them requested to be compliant to this interface)
```

Comment:
we can omit this comment.
are there other phases that have it?

Reply:
I guess that all phases have

Reply:
updated.